### PR TITLE
modules/linux: add support for building with kernel 5.4.69

### PR DESCRIPTION
**Pull request by MrChromebox**
Add support to linux module, port patches from 4.19.139. 

Needed for upcoming platforms (e.g., CometLake-U) not supported by 4.19 kernel.

Migrate all current Librem boards to use the new 5.4.69 kernel, so existing and upcoming boards will all use same kernel/config.

Signed-off-by: Matt DeVillier <matt.devillier@puri.sm>

### CODE REVIEW

File: modules/linux [hunk 1]
```diff
@@ -22,7 +25,11 @@ linux_kconfig := $(or $(CONFIG_LINUX_CONFIG),config/linux.config)
 linux_dir := $(linux_base_dir)/$(notdir $(basename $(linux_kconfig)))
 
 linux_tar := linux-$(linux_version).tar.xz
+ifeq "$(CONFIG_LINUX_VERSION)" "5.4.69"
```
Comment (MrChromebox) [inline modules/linux:28]: probably need a cleaner check than this, especially if we need to support multiple 5.x kernels in the future. open to suggestions.

Comment (tlaurion) [reply modules/linux:28]: @https://stackoverflow.com/questions/2741708/makefile-contains-string ?
Tweaked to begin search at the beginning of literal?

Comment (MrChromebox) [reply modules/linux:28]: @tlaurion ignoring readability,
`linux_url := https://cdn.kernel.org/pub/linux/kernel/v$(basename $(basename $(CONFIG_LINUX_VERSION))).x/$(linux_tar)` should work for all 2-3-digit kernel versions, if I'm reading the makefile docs correctly

**Comment by MrChromebox** [comment]
ok, much cleaner now :)

**Comment by tlaurion** [comment]
@MrChromebox your server board depends on this? Can you create a PR that makes the build so that can of change can at least show compilation and not breaking other boards in the future? Tag me back to ease followup please.

**Comment by MrChromebox** [comment]
@tlaurion no the servers using 4.19.x still, 5.4.x is needed for our upcoming 10th-gen/Comet Lake based devices (Librem 14 laptop and others) which I'm not quite ready to upstream. I could bump the Mini since it's already being built by CI

**Comment by tlaurion** [comment]
@MrChromebox : new tools (rsync) needed on host to build Linux it seems?
Good thing is that documentation points to the CircleCI file now for builder requirements, so a single change in Ci config permits users to understand change in builder's requirements...

**Comment by tlaurion** [comment]
@MrChromebox : 
```
librem_mini/util/cbfstool/cbfstool librem_mini/coreboot.pre.tmp add-payload -f ../../build/librem_mini/bzImage -n fallback/payload  -c none  -r COREBOOT   -C "intel_iommu=igfx_off quiet loglevel=2" -I "../../build/librem_mini/initrd.cpio.xz"
E: Could not add [../../build/librem_mini/bzImage, 7326258 bytes (7154 KB)@0x0]; too big?
E: Failed to add '../../build/librem_mini/bzImage' into ROM image.
E: Failed while operating on 'COREBOOT' region!
E: The image will be left unmodified.
make[1]: *** [Makefile.inc:1020: librem_mini/coreboot.pre] Error 1
make[1]: Leaving directory '/root/project/build/coreboot-4.12'
make: *** [Makefile:438: /root/project/build/coreboot-4.12/librem_mini/.build] Error 1
```

**Comment by tlaurion** [comment]
For the sake of this PR, We now understand that the build of the kernel works while it is too bloated per kernel config for Heads space constraints.

We could leave the kernel module there, remove rsync dependency in circle CI and/or have this merged in when a board depending on that kernel config is tiny enough to fit as a coreboot payload?

**Comment by tlaurion** [comment]
@MrChromebox 
From https://app.circleci.com/pipelines/github/MrChromebox/heads/33/workflows/20d21376-8bce-4371-b3f5-2ad870c58895/jobs/36

```
librem_mini/util/cbfstool/cbfstool librem_mini/coreboot.pre.tmp add-payload -f ../../build/librem_mini/bzImage -n fallback/payload  -c none  -r COREBOOT   -C "intel_iommu=igfx_off quiet loglevel=2" -I "../../build/librem_mini/initrd.cpio.xz"
E: Could not add [../../build/librem_mini/bzImage, 7323186 bytes (7151 KB)@0x0]; too big?
E: Failed to add '../../build/librem_mini/bzImage' into ROM image.
E: Failed while operating on 'COREBOOT' region!
E: The image will be left unmodified.
make[1]: *** [Makefile.inc:1020: librem_mini/coreboot.pre] Error 1
make[1]: Leaving directory '/root/project/build/coreboot-4.12'
```
**TIP: Open CI log, CTRL-F and search for `<==`. First occurence delimits between (up) general make $BOARD statement (up) where individual logs that were modified in the last moinute are outputted (after)**

**Comment by MrChromebox** [comment]
will be easier if I just push this change as part of the addition of the board that uses it, as there won't be any issues with CBFS size etc.

**Comment by MrChromebox** [comment]
hmm, confused as to why it's still trying to build against kernel 4.19.x

**Comment by tlaurion** [comment]
@MrChromebox :
```
printf "    CBFS       fallback/payload\n"
    CBFS       fallback/payload
librem_mini/util/cbfstool/cbfstool librem_mini/coreboot.pre.tmp add-payload -f ../../build/librem_mini/bzImage -n fallback/payload  -c none  -r COREBOOT   -C "intel_iommu=igfx_off quiet loglevel=2" -I "../../build/librem_mini/initrd.cpio.xz"
E: Could not add [../../build/librem_mini/bzImage, 7360562 bytes (7188 KB)@0x0]; too big?
E: Failed to add '../../build/librem_mini/bzImage' into ROM image.
E: Failed while operating on 'COREBOOT' region!
E: The image will be left unmodified.
make[1]: *** [Makefile.inc:1020: librem_mini/coreboot.pre] Error 1
rm librem_mini/util/cbfstool/fmd_parser.c librem_mini/util/cbfstool/fmd_scanner.c
make[1]: Leaving directory '/root/project/build/coreboot-4.12'
make: *** [Makefile:438: /root/project/build/coreboot-4.12/librem_mini/.build] Error 1
```

Where server boards built with 5.4 kernel https://circleci.com/api/v1.1/project/github/tlaurion/heads/585/output/107/0?file=true&allocation-id=5f943e41ea037f6e68b9a264-0-build%2F7D27093

**Comment by MrChromebox** [comment]
@tlaurion I swore I'd rebased it but I guess not. Done now

**Pull request merged**